Task3/triangle.go

```diff
@@ -0,0 +1,52 @@
+// Package triangle contains mathods to detect what kind given triangle is.
+//
+package triangle
+
+import (
+	"math"
+)
+
+// Kind triangle type representation
+type Kind int
+
+const (
+	// NaT is not a triangle
+	NaT = iota
+	// Equ equilateral
+	Equ
+	// Iso isosceles
+	Iso
+	// Sca scalene
+	Sca
+	// Deg degenerate triangle
+	Deg
+)
+
+func KindFromSides(a float64, b float64, c float64) Kind {
+	if math.IsNaN(a) == true || math.IsNaN(b) == true || math.IsNaN(c) == true {
+		return NaT
+	}
+	if a == math.Inf(1) || b == math.Inf(1) || c == math.Inf(1) {
+		return NaT
+	}
+	if a == math.Inf(-1) || b == math.Inf(-1) || c == math.Inf(-1) {
+		return NaT
+	}
+	if a <= 0 || b <= 0 || c <= 0 {
+		return NaT
+	}
+	if a+b < c || b+c < a || a+c < b {
+		return NaT
+	}
+	if a+b == c || b+c == a || a+c == b {
+		return Deg
+	}
+	if a == b && b == c {
+		return Equ
+	}
+	if a == b || b == c || a == c {
+		return Iso
+	} else {
+		return Sca
+	}
+}
```

Comment:
Looks good, but some tests are failed. It seems the bug in the tests. I've looked into the problem and found that tests doesn't expect degenerated triangles. You can examine test suite and find out that we don't have any test case with degenerate triangles. And if you go further, and read the readme file carefully, you can see that this situation is described there: 
```
The case where the sum of the lengths of two sides _equals_ that of the
third is known as a _degenerate_ triangle - it has zero area and looks like
a single line. Feel free to add your own code/tests to check for degenerate triangles.
```
**Feel free to add your own code/tests to check for degenerate triangles** but it is not necessary.